### DISCUSSION TITLE
HTTP vs MQTT (replicating approach to other devices?)

### DISCUSSION THREAD

**Discussion by donparlor** [krahabb/meross_lan#162] [Q&A]
Hello @krahabb,

I have a stupid question, but while reading your [Wiki page](https://github.com/krahabb/meross_lan/wiki/HTTP-vs-MQTT), I wondered if your approche to meross_lan could be replicated with other brand devices. I am trying to find a way to integrate my 4" Bazz Smart recessed RGB lights to HA, but the only option seems to be flashing [Tasmota](https://templates.blakadder.com/bazz_WFDISK400W.html) in the TYWE2S controller of these devices. 

I know your time is precious, and I'm not asking you to do it, but where would be the starting point to even know if this could be possible? What is the first step if anyone want to try it?

PS: Maybe this question should not be asked here, but I don't know how to contact directly.

Thanks

**Comment by krahabb** [comment]
Hello @donparlor ,
It's hard to set a general rule when you come to IoT devices.
Meross devices have both an HTTP interface and an MQTT one. Moreover these interfaces are interachangeable when it comes to general device features while, for example, the device pairing and configuration is mainly available only on HTTP (this is likely due to the fact that MQTT need some configuration in order to work on the device itself while HTTP is readily avaialable as soon as you power it up with any internal configuration.
This is not a general rule so I hardly expect any other brand device to be able to work 'seamlessly' on both protocols. There is no general standard in IoT except, some brands share a common technological infrastructure and so they all interface with the same protocols (see Tuya as probably the most prominent platform)

**Reply by donparlor** [reply]
It seems software developers at Bazz used Tuya IoT Dev Platform. 
![unnamed](https://user-images.githubusercontent.com/97048708/151500054-36bb2c34-1275-4269-96ab-2d517bd330fb.png)
<img width="1287" alt="Capture d’écran, le 2022-01-22 à 11 13 28" src="https://user-images.githubusercontent.com/97048708/151500065-e89dda1b-4bea-4ad3-ba40-8630cf0f17b6.png">

Does Tuya IoT Dev Platform would allow to replicate a similar integration as you did with meross_lan or not?

**Reply by krahabb** [reply]
Tuya devices are now efficiently supported by the [official Tuya integration](https://www.home-assistant.io/integrations/tuya/) which has recently been improved. I don't know the details i.e. if it's capable of local only connection or if it only works on cloud. For sure it is well integrated and provides an easy solution for all of these brands implementing tuya protocol.
Replicating something like meross_lan for tuya devices is not as easy but I cannot really tell since I don't really know how the tuya protocol works.
If you're in for the job and the hardware supports it I would definitely re-flash the device with something like Tasmota or, better, EspHome which provides a native lan only integration with HA

**Reply by donparlor** [reply]
Thank you @krahabb! 

I don't think I have the skills to hack into Tuya protocols to create a local connection with those devices. I think many skilled programmers have already tried this but could not come up with a solution without re-flashing Tuya devices. Furthermore, Bazz Smart might have other perks that would make it harder to bypass. I think I will try to Flash Tasmota as there is already a template for my device within Tasmota repository. I would have preferred to use the App + HA at the same time for these devices but I bet I won't miss Bazz App.

Thanks again. and many thanks for your time with Meross_Lan! It's really a great add-on!
